lib/phoenix_live_view/test/client_proxy.ex

```diff
@@ -1001,8 +1003,33 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
     end
   end
 
-  defp maybe_js_event("[" <> _ = encoded_js) do
-    js = encoded_js |> DOM.parse() |> Phoenix.json_library().decode!()
+  defp maybe_decode_js("[" <> _ = encoded_js),
+    do: {:decoded, encoded_js |> DOM.parse() |> Phoenix.json_library().decode!()}
+
+  defp maybe_decode_js(event), do: {:raw, event}
+
+  defp check_not_patch_or_navigate({:decoded, js}, topic) do
+    op = Enum.filter(js, fn [kind, _args] -> kind == "patch" || kind == "navigate" end)
```

Comment:
There can be multiple ops so will need to Enum.find here I believe

Reply:
Multiple valid operations within a single message? I modelled this on the push support below which only supports one such event at a time, and it makes sense to me to also only support one patch or navigate at a time, especially navigate, so using find would remove the ability to raise an error here if two such events (like patch then navigate) are detected?

Reply:
Isn't that what this does, filter the events down to only navigate / patch and then pattern matches on the result checking theres only one event of that kind?

Reply:
I also don't see how using find would help here. It'd be great if this feature landed. Maybe you can take another look? @chrismccord